libsrc/grabber/v4l2/V4L2Grabber.cpp

```diff
@@ -71,11 +80,66 @@ V4L2Grabber::V4L2Grabber(const QString & device
 	setInput(input);
 	setWidthHeight(width, height);
 	setFramerate(fps);
-	setDeviceVideoStandard(device, videoStandard);
+	setDeviceVideoStandard(device, videoStandard);	
+}
+
+void V4L2Grabber::loadLutFile(QString path)
+{
+	//load lut
+	QString fileName3d = path +"/lut_rgb.3d";
+	
+	if (lutBuffer!=NULL)
+		free(lutBuffer);
+	lutBuffer = NULL;
+	
+	if (FILE *file = fopen(QSTRING_CSTR(fileName3d), "r")) {
+		size_t length;
+		Debug(_log,"LUT file found: %s", QSTRING_CSTR(fileName3d));
+		
+		fseek(file, 0, SEEK_END);
+		length = ftell(file);
+		fseek(file, 0, SEEK_SET);
+		if (length ==  LUT_FILE_SIZE) {
+			lutBuffer = (unsigned char *)malloc(length + 1);
+			if(fread(lutBuffer, 1, length, file)!=length)
+			{
+				free(lutBuffer);
+				lutBuffer = NULL;
+				Debug(_log,"Error reading LUT file");
+			}
+			else
+				Debug(_log,"LUT file has been loaded");
+		}
+		else
+			Debug(_log,"LUT file has invalid length: %i", length);
+		fclose(file);
+	}
+	else
+		Debug(_log,"LUT file NOT found: %s", QSTRING_CSTR(fileName3d));
+		
+}
+
+void V4L2Grabber::setFpsSoftwareDecimation(int decimation)
+{
+	_fpsSoftwareDecimation = decimation;
+	Debug(_log,"setFpsSoftwareDecimation to: %i", decimation);
+}
+
+void V4L2Grabber::setHdrToneMappingEnabled(bool enable)
+{
+	_hdrToneMappingEnabled = enable;
+	if (lutBuffer!=NULL || !enable)
+		Debug(_log,"setHdrToneMappingEnabled to: %s", enable ? "true" : "false");
+	else
+		Error(_log,"ERROR! setHdrToneMappingEnabled was to: enable, but the 3DLUT file (lut_rgb.3d)  was missing in the configuration folder");		
 }
 
 V4L2Grabber::~V4L2Grabber()
 {
+	if (lutBuffer!=NULL)
```

Comment:
no need for check, free(nullptr) is safe

Reply:
ok